docs/07_how-to/14_ht-front-end-simple.mdx

```diff
@@ -0,0 +1,273 @@
+    
+    export const newPageStyles = css`
+      // add your styles here
+
+```
+
+    - **new-page.template.ts** (this is where your HTML template is defined)
+
+```ts
+    import { Charts } from './charts';
+    
+    export const newPageTemplate = html`
+       <div>This is the new page template!</div>
+    `
+    ```
+
+3. In the file **src/routes/config.ts**, create a route for your new page, for example:
+
+```TypeScript
+{
+  path: 'new-page',
+  element: NewPage,
+  title: 'New Page',
+  name: 'new-page',
+  navItems: [
+    {
+      title: 'New Page',
+      icon: {
+        name: 'cog', // // use any free icon from https://fontawesome.com/icons
+        variant: 'solid',
+      },
+      permission: '',
+    },
+  ],
+},
+```
+
+## Adding a form
+You have created a component called `NewPage`. Now you can add a form to that component. For example, you would use a form to add a record to a table in the database.
+
+The example below adds a simple form element to the new page. 
+
+- The content of the form is defined by the UISchema and the JsonSchema in the **new-page.ts** file.
+- The form is itself is then added to the **new-page.template.ts** file.
+
+**new-page.ts**
+
+```ts
+import { customElement, FoundationElement, observable } from '@genesislcap/web-core';
+...
+export class NewPage extends FoundationElement {
+  ...
+  formUiSchema = {
+    type: 'VerticalLayout',
+    elements: [
+      {
+        type: 'Control',
+        scope: '#/properties/QUANTITY',
+        label: 'Enter Quantity:'
+      },
+      {
+        type: 'Control',
+        scope: '#/properties/SIDE',
+        options: <ConnectedRenderersOptions>{
+          data: [
+            { label: 'Buy', value: 'BUY' },
+            { label: 'Sell', value: 'SELL' }
+          ],
+          valueField: 'value',
+          labelField: 'label',
+        }
+      }
+    ]
+  }
+  
+  formJsonSchema = {
+    type: 'object',
+    properties: {
+      quantity: {
+        type: 'number',
+        description: 'kotlin.Double',
+      },
+      side: {
+        type: 'number',
+        description: 'kotlin.String',
+      },
+    },
+  };
+
+  handleFormSubmit(e: CustomEvent) {
+    // handle form submit
+  }
+  ...
+}
+```
+
+**new-page.template.ts**
+
+```ts
+...
+<foundation-form 
+  @submit=${(x, ctx) => x.handleFormSubmit(ctx.event)} 
+  :jsonSchema=${x => x.formJsonSchema} 
+  :uischema=${x => x.formUiSchema}
+>
+</foundation-form>
+...
+```
+
+## Adding a modal
+To create a modal component, add the element to your **new-page.template.ts** file. The component, and any content, will be invisible until the modal is opened.
+
+In the **new-page.ts** class, add the property to get a handle on the modal instance.
+
+```
+export class NewPage extends FoundationElement {
+  myModal: Modal;
+}
+```
+
+In the **new-page.template.ts** add the element and a `ref` value so that the class can open it.
+
+In the click handler on the button, call the `show` method on the modal.
```

Comment:
Not today necessarily but could we show a modal pop up after an action, with some context, which is more likely to help people out on a more frequent basis?

Reply:
Maybe its in the example application? If so could we allude to the use case in summary?